packages/squiggle-lang/src/reducer/lambda.ts

```diff
@@ -164,6 +166,10 @@ export class BuiltinLambda extends BaseLambda {
   ) {
     super((args, context) => this._call(args, context));
     this._definitions = signatures;
+
+    // TODO - this sets the flag that the function is a decorator, but later it we don't check which signatures are decorators.
```

Comment:
later **if** we don't check

Reply:
"it" in the comment was an extra accidental word, sorry.

The problem is this: if we have a function with multiple definitions, and only some of those definitions were marked as decorators, then it'd be enough for the entire lambda to be marked as a possible decorator (because I check `signatures.some`; maybe I should've used `signatures.every` instead).

But when this decorator gets applied, there's no check that the individual _definition_ that has matched the parameters is a decorator.

I could fix this by passing "called as a decorator" around, but the chain of calls is longer than I'd like (reducer -> `BaseLambda.callFrom` -> `BuiltinLambda._call` -> `tryCallFnDefinition`, and then filter out the definitions that can be decorators), and it would have (very minor) performance consequences.

This can be fixed in the future when we implement a separate namespace for decorators.